Error in log from missing metadata.json

Briefcase on the CLI reports an error: 

```
Unexpected error in Briefcase. Please review briefcase.log for more information. For help, post to https://forum.opendatakit.org/c/support
```

The log has
```
java.io.UncheckedIOException: java.nio.file.NoSuchFileException: uga/ODK Briefcase Storage/uga/ODK Briefcase Storage/forms/IOMHHSurveyer_V6/metadata.json
	at org.opendatakit.briefcase.reused.UncheckedFiles.write(UncheckedFiles.java:80)
	at org.opendatakit.briefcase.reused.UncheckedFiles.write(UncheckedFiles.java:73)
	at org.opendatakit.briefcase.model.form.FileSystemFormMetadataAdapter.serialize(FileSystemFormMetadataAdapter.java:123)
	at org.opendatakit.briefcase.model.form.FileSystemFormMetadataAdapter.persist(FileSystemFormMetadataAdapter.java:96)
	at java.base/java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:441)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1621)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.opendatakit.briefcase.model.form.FileSystemFormMetadataAdapter.syncWithFilesAt(FileSystemFormMetadataAdapter.java:75)
	at org.opendatakit.briefcase.model.form.FileSystemFormMetadataAdapter.at(FileSystemFormMetadataAdapter.java:33)
	at org.opendatakit.briefcase.operations.PullFormFromAggregate.pullFormFromAggregate(PullFormFromAggregate.java:90)
	at org.opendatakit.briefcase.operations.PullFormFromAggregate.lambda$static$0(PullFormFromAggregate.java:69)
	at org.opendatakit.common.cli.Cli.lambda$run$4(Cli.java:130)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at org.opendatakit.common.cli.Cli.run(Cli.java:127)
	at org.opendatakit.briefcase.Launcher.main(Launcher.java:90)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.eclipse.jdt.internal.jarinjarloader.JarRsrcLoader.main(JarRsrcLoader.java:58)
Caused by: java.nio.file.NoSuchFileException: uga/ODK Briefcase Storage/uga/ODK Briefcase Storage/forms/IOMHHSurveyer_V6/metadata.json
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
	at java.base/java.nio.file.Files.newOutputStream(Files.java:223)
	at java.base/java.nio.file.Files.write(Files.java:3488)
	at org.opendatakit.briefcase.reused.UncheckedFiles.write(UncheckedFiles.java:78)
	... 23 common frames omitted
```

--- a/src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java
+++ b/src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java
@@ -23,6 +23,7 @@ import java.util.stream.Stream;
 import org.opendatakit.briefcase.export.XmlElement;
 import org.opendatakit.briefcase.reused.BriefcaseException;
 import org.opendatakit.briefcase.reused.LegacyPrefs;
+import org.opendatakit.briefcase.reused.UncheckedFiles;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -130,21 +131,12 @@ public class FileSystemFormMetadataAdapter implements FormMetadataPort {
   private static Path serialize(FormMetadata metaData) {
     try {
       Path formDir = metaData.getFormDir();
-      if (Files.exists(formDir)) {
-        return write(
-            getMetadataFile(metaData),
-            MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(metaData.asJson(MAPPER)),
-            CREATE, TRUNCATE_EXISTING
-        );
-      } else {
-        Files.createDirectory(formDir);
-        return write(
-            getMetadataFile(metaData),
-            MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(metaData.asJson(MAPPER)),
-            CREATE, TRUNCATE_EXISTING
-        );
-      }
-
+      UncheckedFiles.createDirectories(formDir);
+      return write(
+          getMetadataFile(metaData),
+          MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(metaData.asJson(MAPPER)),
+          CREATE, TRUNCATE_EXISTING
+      );
     } catch (IOException e) {
       throw new BriefcaseException("Couldn't produce JSON FormMetadata", e);
     }

--- a/src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java
+++ b/src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java
@@ -65,6 +65,7 @@ public class FileSystemFormMetadataAdapter implements FormMetadataPort {
     // from actually writing files in the same folders we're reading from
     List<FormMetadata> metadataFiles = formFiles.map(formFile -> {
       Path formDir = formFile.getParent();
+      //System.out.println("formFile.getParent(): "+formDir);
       Path metadataFile = formDir.resolve("metadata.json");
       FormMetadata formMetadata = Files.exists(metadataFile) ? deserialize(storageRoot, metadataFile) : FormMetadata.from(storageRoot, formFile);
       if (!formMetadata.getCursor().isEmpty())
@@ -132,12 +133,23 @@ public class FileSystemFormMetadataAdapter implements FormMetadataPort {
 
   private static Path serialize(FormMetadata metaData) {
     try {
-      return write(
-          getMetadataFile(metaData),
-          MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(metaData.asJson(MAPPER)),
-          CREATE, TRUNCATE_EXISTING
-      );
-    } catch (JsonProcessingException e) {
+      Path formDir = metaData.getFormDir();
+      if (Files.exists(formDir)) {
+        return write(
+            getMetadataFile(metaData),
+            MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(metaData.asJson(MAPPER)),
+            CREATE, TRUNCATE_EXISTING
+        );
+      } else {
+        Files.createDirectory(formDir);
+        return write(
+            getMetadataFile(metaData),
+            MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(metaData.asJson(MAPPER)),
+            CREATE, TRUNCATE_EXISTING
+        );
+      }
+
+    } catch (IOException e) {
       throw new BriefcaseException("Couldn't produce JSON FormMetadata", e);
     }
   }
@@ -151,11 +163,7 @@ public class FileSystemFormMetadataAdapter implements FormMetadataPort {
   }
 
   private static Path getMetadataFile(FormMetadata metaData) {
-    try {
-      return metaData.getFormDir().resolve("metadata.json");
-    } catch (InvalidPathException e) {
-      throw new BriefcaseException("cannot resolve metadata path", e);
-    }
+    return metaData.getFormDir().resolve("metadata.json");
   }
   // endregion
 }

--- a/src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java
+++ b/src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java
@@ -7,14 +7,11 @@ import static java.util.stream.Collectors.toMap;
 import static org.opendatakit.briefcase.reused.UncheckedFiles.walk;
 import static org.opendatakit.briefcase.reused.UncheckedFiles.write;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.nio.file.Files;
-import java.nio.file.InvalidPathException;
-import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.util.List;
 import java.util.Map;
@@ -65,7 +62,6 @@ public class FileSystemFormMetadataAdapter implements FormMetadataPort {
     // from actually writing files in the same folders we're reading from
     List<FormMetadata> metadataFiles = formFiles.map(formFile -> {
       Path formDir = formFile.getParent();
-      //System.out.println("formFile.getParent(): "+formDir);
       Path metadataFile = formDir.resolve("metadata.json");
       FormMetadata formMetadata = Files.exists(metadataFile) ? deserialize(storageRoot, metadataFile) : FormMetadata.from(storageRoot, formFile);
       if (!formMetadata.getCursor().isEmpty())

--- a/src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java
+++ b/src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java
@@ -13,6 +13,8 @@ import com.fasterxml.jackson.databind.ObjectMapper;
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.util.List;
 import java.util.Map;
@@ -149,7 +151,11 @@ public class FileSystemFormMetadataAdapter implements FormMetadataPort {
   }
 
   private static Path getMetadataFile(FormMetadata metaData) {
-    return metaData.getFormDir().resolve("metadata.json");
+    try {
+      return metaData.getFormDir().resolve("metadata.json");
+    } catch (InvalidPathException e) {
+      throw new BriefcaseException("cannot resolve metadata path", e);
+    }
   }
   // endregion
 }

